app/controllers/admin/users_controller.rb

```diff
@@ -97,10 +97,15 @@ def change_salesforce_contact
 
       begin
         contact = OpenStax::Salesforce::Remote::Contact.find(new_id) if check_really_exists
-        # if didn't explode, we found it, let cron job update other info
-        @user.salesforce_contact_id = new_id
-        return @user.save
+
+        if contact.present? || new_id.nil?
+          @user.salesforce_contact_id = new_id
+          return @user.save
+        end
       rescue
+        # exploded, probably due to badly formed SF ID
+      ensure
+        # either exploded or contact was `nil`
         flash[:alert] = "Can't find a Salesforce contact with ID #{new_id}"
```

Comment:
This block tricked me up. I first thought that since you're using `find` for the contact, it will always be present or will have raised a not found exception.  But then I realized I'd missed the `if check_really_exists` guard. 

Maybe it'd be clearer as:
```ruby
if check_really_exists && new_id.present?
  contact = OpenStax::Salesforce::Remote::Contact.find(id: new_id)
  @user.salesforce_contact_id = new_id
  return @user.save
end
rescue # catch NotFound or whatever
...
```

Also, the `ensure` block will always be executed, so it will populate the `flash` even if the `return @user.save` executed.  I think it'll need a `unless @user && @user.valid?` tacked onto the end.